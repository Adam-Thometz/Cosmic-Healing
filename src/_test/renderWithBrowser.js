import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

function withProvider({ children, initialRoutes }) {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
)
};

function renderWithBrowser(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithBrowser;