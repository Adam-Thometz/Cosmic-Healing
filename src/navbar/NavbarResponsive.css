.NavbarResponsive {
  display: flex;
  justify-content: space-between;
  width: 95%;
  margin: 16px auto;
  position: relative;
}

.NavbarResponsive-nav {
  display: flex;
  flex-direction: column;
}

.NavbarResponsive-nav img {
  width: 18px;
  height: 12px;
}

.NavbarResponsive-links {
  display: flex;
  flex-direction: column;
  text-align: left;
  padding: 8px;
  background-color: white;
  margin: 8px 0;
  position: absolute;
  top: 16px;
  left: 0;
  z-index: 5;
  border: 1px solid black;
  border-radius: 4px;
}

.NavbarResponsive-links a:not(a:last-child) {
  padding: 12px 0;
  color: #333333;
  text-decoration: none;
  font-size: 18px;
  font-family: Nuckle;
  font-weight: 700;
  line-height: 16px;
  border-bottom: 1px solid #CFC8D8;
}

.NavbarResponsive-links a.active {
  color: #D87B41;
}