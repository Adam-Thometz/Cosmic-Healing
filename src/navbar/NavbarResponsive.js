import React, { useRef } from 'react';
import { NavLink } from 'react-router-dom';

import './NavbarResponsive.css';

import Link from '../_components/link/Link';

import hamburger from '../_media/hamburger.png';

const NavbarResponsive = () => {
  const links = useRef();
  const active = ({ isActive }) => isActive ? 'active' : '';
  
  const showLinks = () => {
    links.current.style.display = 
      links.current.style.display === 'none' ? 'flex' : 'none'
  }

  return (
    <header className='NavbarResponsive'>
      <nav className='NavbarResponsive-nav'>
        <img src={hamburger} alt="Click for navigation" onClick={showLinks} />
        <div style={{ display: 'none' }} ref={links} className='NavbarResponsive-links'>
          <NavLink to='/' className={active} end>Home</NavLink>
          <NavLink to='about-me' className={active}>About Me</NavLink>
          <NavLink to='services' className={active}>Services</NavLink>
          <Link primary>Book a free session</Link>
        </div>
      </nav>
      <img src={''} alt='' />
    </header>
  );
};

export default NavbarResponsive;