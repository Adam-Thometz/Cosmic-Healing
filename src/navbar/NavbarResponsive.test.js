import React from "react";
import renderWithBrowser from '../_test/renderWithBrowser';
import NavbarResponsive from './NavbarResponsive';

describe('NavbarResponsive component', () => {
  it('renders without crashing', () => {
    renderWithBrowser(<NavbarResponsive />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = renderWithBrowser(<NavbarResponsive />);
    expect(asFragment()).toMatchSnapshot();
  });
});