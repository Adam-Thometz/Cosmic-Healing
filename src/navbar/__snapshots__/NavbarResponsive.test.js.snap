// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarResponsive component matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="NavbarResponsive"
  >
    <nav
      class="NavbarResponsive-nav"
    >
      <img
        alt="Click for navigation"
        src="hamburger.png"
      />
      <div
        class="NavbarResponsive-links"
        style="display: none;"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          class=""
          href="/about-me"
        >
          About Me
        </a>
        <a
          class=""
          href="/services"
        >
          Services
        </a>
        <a
          class="Link primary"
        >
          Book a free session
        </a>
      </div>
    </nav>
    <img
      alt=""
      src=""
    />
  </header>
</DocumentFragment>
`;
